metricbeat/module/kibana/stats/stats.go

```diff
@@ -75,26 +77,17 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 func (m *MetricSet) Fetch(r mb.ReporterV2) error {
 	err := m.init()
 	if err != nil {
-		if m.XPackEnabled {
-			m.Logger().Error(err)
-			return nil
-		}
 		return err
 	}
 
 	now := time.Now()
 
-	err = m.fetchStats(r, now)
-	if err != nil {
-		if m.XPackEnabled {
-			m.Logger().Error(err)
-			return nil
-		}
-		return err
+	if err = m.fetchStats(r, now); err != nil {
+		return errors.Wrap(err, "error trying to get stats data from Kibana")
 	}
 
-	if m.XPackEnabled {
-		m.fetchSettings(r, now)
+	if err = m.fetchSettings(r); err != nil {
+		return errors.Wrap(err, "error trying to get settings data from Kibana")
```

Comment:
So previously (before this PR), this metricset would behave as follows:
- if `xpack.enabled: true` was set, it would produce two events in each `Fetch` iteration, indexed into `.monitoring-kibana-*`, one with `type: kibana_stats` and one with `type: kibana_settings`.
- if `xpack.enabled: false` was set, it would produce one event in each `Fetch` iteration, indexed into `metricbeat-*, corresponding to `type: kibana_stats` (but not actually including that field).

With the changes in this PR, it seems like we would _always_ be producing two events per `Fetch` iteration: one corresponding to `type: kibana_stats` and one corresponding to `type: kibana_settings`. I wonder if now it makes sense to keep the `kibana/stats` metricset for just the former and create a new metricset, `kibana/settings`, for just the latter? It would be more in line with how other Metricbeat modules work. WDYT?

Reply:
Feel free to perform this split of metricsets in a follow up PR to the feature branch. For now, if you want to merge the PR with the current implementation, that's fine.

Reply:
I'll do it in a different PR because I have seen that it requires that I change quite a lot of thing in the base module. I mean, I have to change them anyways but I discovered it now.